### PR TITLE
Add a GDScript template for `VisualShaderNodeCustom`

### PR DESCRIPTION
**Pull request by Chaosus**
As a continuation of #53957 I think it's not bad to add such a template for the visual shader plugin.

### CODE REVIEW

**Comment by fabriceci** [comment]
good idea! Just 2 remarks:

- GDscript templates have to use types by default, the type will be stripped if needed
- Now, all the template exists in GdScript, exists in C#, it would be great to add also one for c# to keep the consistency.

**Comment by Chaosus** [comment]
@fabriceci I think I'm done with it - custom visual shader nodes still doesn't actually work in C# currently, but that is another story

File: modules/mono/editor_templates/VisualShaderNodeCustom/basic.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+public partial class VisualShaderNode_CLASS_ : _BASE_
+{
+    public override string _GetName()
+    {
+        return "_CLASS_";
+    }
+
+	public override string _GetCategory()
+    {
```
Comment (aaronfranke) [inline modules/mono/editor_templates/VisualShaderNodeCustom/basic.cs:14]: The brace indent is off by one on several lines.

File: modules/gdscript/editor_templates/VisualShaderNodeCustom/basic.gd [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+func _get_output_port_name(port : int) -> String:
+	return "result"
+
+func _get_output_port_type(port : int) -> int:
+	return PORT_TYPE_SCALAR
+
+func _get_global_code(mode : Shader.Mode) -> String:
+	return ""
+
+func _get_code(input_vars : Array[String], output_vars : Array[String], mode : Shader.Mode, type : VisualShader.Type) -> String:
```
Comment (akien-mga) [inline modules/gdscript/editor_templates/VisualShaderNodeCustom/basic.gd:40]: I think this should follow the GDScript style guide, so there shouldn't be a space before `:` for the type hints (here and in other methods with parameters): https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_styleguide.html#static-typing
```suggestion
func _get_code(input_vars: Array[String], output_vars: Array[String], mode: Shader.Mode, type: VisualShader.Type) -> String:
```

**Review by akien-mga — APPROVED** [review]

**Pull request merged**